src/Domain/MandateCancellationType.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace MatchBot\Domain;
+
+enum MandateCancellationType: string
```

Comment:
not sure if introducing this enum and associated column on mandate table makes sense. Could instead have split the existing `\MatchBot\Domain\MandateStatus::Cancelled` case into three to discriminate between the three possibilities, and could still map them all to 'Cancelled'  the same thing when sending to SF and FE.

Reply:
OK will keep as is. For clarity this is adding both the enum for the general type of cancellation (e.g. donor cancelled vs initial payment failed) as well as a free text field for any reason given by the donor. I think there will be some business logic that will depend on the distinction, e.g. when donor looks at their list of mandates we'd want to show ones they cancelled but not ones we auto-cancelled. Although we could say that logic is only show mandates that have at least 1 successful donation each.